src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,404 @@
+        }
+
+        sliderA.HintTooltip = TranslationServer.Translate("COLOUR_PICKER_A_TOOLTIP");
+    }
+
+    private void UpdateButtonsState()
+    {
+        if (hsvCheckButton == null)
+            return;
+
+        hsvCheckButton.Pressed = HsvMode;
+        rawCheckButton.Pressed = RawMode;
+
+        hsvCheckButton.Disabled = !hsvButtonEnabled || RawMode;
+        rawCheckButton.Disabled = !rawButtonEnabled || HsvMode;
+    }
+
+    private void OnAddPresetButtonPressed()
+    {
+        if (!presetsEnabled)
+            return;
+
+        AddPreset(Color);
+    }
+
+    private void OnHSVButtonToggled(bool isOn)
+    {
+        HsvMode = isOn;
+    }
+
+    private void OnRawButtonToggled(bool isOn)
+    {
+        RawMode = isOn;
+    }
+
+    private void OnColourChanged(Color colour)
+    {
+        htmlColourEdit.Text = colour.ToHtml();
+    }
+
+    /// <summary>
+    ///   Called when (keyboard) entered in HtmlColourEdit or from OnHtmlColourEditFocusExited.
+    ///   Set Color when text is valid; reset if not.
+    /// </summary>
+    /// <param name="colour">Current htmlColourEditor text</param>
+    private void OnHtmlColourEditEntered(string colour)
+    {
+        if (colour.IsValidHtmlColor())
+        {
+            Color = new Color(colour);
+        }
+        else
+        {
+            htmlColourEdit.Text = Color.ToHtml();
+        }
+    }
+
+    /// <summary>
+    ///   Called when focus exited HtmlColourEdit.
+    /// </summary>
+    private void OnHtmlColourEditFocusExited()
+    {
+        OnHtmlColourEditEntered(htmlColourEdit.Text);
+        htmlColourEdit.Deselect();
+    }
+
+    private class TweakedColourPickerPreset : ColorRect
+    {
+        private readonly TweakedColourPicker owner;
+
+        public TweakedColourPickerPreset(TweakedColourPicker owner, Color colour)
+        {
+            this.owner = owner;
+            Color = colour;
+            MarginTop = MarginBottom = MarginLeft = MarginRight = 6.0f;
+            RectMinSize = new Vector2(20, 20);
+            SizeFlagsHorizontal = 8;
+            SizeFlagsVertical = 4;
+            UpdateTooltip();
+        }
+
+        public override void _Notification(int what)
+        {
+            if (what == NotificationTranslationChanged)
+                UpdateTooltip();
+
+            base._Notification(what);
+        }
+
+        public override void _GuiInput(InputEvent inputEvent)
+        {
+            if (inputEvent is InputEventMouseButton { Pressed: true } mouseEvent)
+            {
+                switch ((ButtonList)mouseEvent.ButtonIndex)
+                {
+                    case ButtonList.Left:
+                        owner.Color = Color;
+                        break;
+                    case ButtonList.Right:
+                        if (!owner.PresetsEnabled)
+                            break;
+
+                        owner.ErasePreset(Color);
```

Comment:
Shouldn't these call the Godot consume input `GetTree().SetInputAsHandled()` and return because these are active actions that are performed? The mouse input shouldn't be able to cause any further actions, that would be a bit sily.